python/src/aiconfig/editor/server/server_utils.py

```diff
@@ -236,6 +235,20 @@ def init_server_state(app: Flask, edit_config: EditServerConfig) -> Result[None,
             return core_utils.ErrWithTraceback(e)
 
 
+def _create_new_aiconfig() -> AIConfigRuntime:
+    aiconfig_runtime = AIConfigRuntime.create()  # type: ignore
+    model_ids = ModelParserRegistry.parser_ids()
+    if len(model_ids) > 0:
+        LOGGER.info(f"{model_ids=}")
+        for model_id in model_ids:
+            if aiconfig_runtime.metadata.models is None:
+                aiconfig_runtime.metadata.models = {}
+            if model_id not in aiconfig_runtime.metadata.models:
+                aiconfig_runtime.add_model(model_id, {"model": model_id})
```

Comment:
Isn't this going to set {"model"} into the completion params for all models by default now? For example, for HuggingFaceTextGenerationParser the settings would have {"model": "HuggingFaceTextGenerationParser"} in this case? I don't think that's what we want, right?

Isn't this issue only a problem for those model parsers that actually support / need the 'model' in their completion params? If so, I think we should resolve this at the model parser level.

e.g. for openai deserialize:
```
model_settings = self.get_model_settings(prompt, aiconfig) or {"model": aiconfig.get_model_name(prompt)})
```

cc @jonathanlastmileai , @rossdanlm

Reply:
Right, good point. I'm a bit confused again. Let's start from here: let's write a minimum example of a *valid* serialized AIConfig for GPT4, and another valid serialized AIConfig for a model that doesn't put a "model" key into the underlying API - maybe HF text generation. Let's get crystal clear on why each one is valid and why alternatives would be invalid.

GPT4

```
{
  "metadata": {
    "models": {
      "gpt-4": {
        "model": "gpt-4"
      }
    }
  },
  "prompts": [
    {
      "name": "get_activities",
      "input": "Tell me 10 fun attractions to do in NYC."
      "metadata": {
        "model": "gpt-4",
      }
    }
  ]
}
```

I think this is valid because the string-valued prompt.metadata.model points to an existing metadata.models key whose value is a dict.

This is an *invalid* variant of what we have above, what we were seeing before:


```
{
  "metadata": {
    "models": {
    }
  },
  "prompts": [
    {
      "name": "get_activities",
      "input": "Tell me 10 fun attractions to do in NYC."
      "metadata": {
        "model": "gpt-4",
      }
    }
  ]
}
```

Right so far?

Reply:
Assuming that's correct, here's something I'm not clear on: what is a valid and invalid equivalent for another model e.g. HuggingFaceTextGenerationParser ? 

By the design of AIConfig, shouldn't it just be the same as for GPT4 but with the name changed?